sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadResourceServerWebSecurityConfigurerAdapter.java

```diff
@@ -30,10 +36,17 @@ protected void configure(HttpSecurity http) throws Exception {
         // @formatter:off
         http.oauth2ResourceServer()
             .jwt()
-            .jwtAuthenticationConverter(
-                new AadJwtBearerTokenAuthenticationConverter(
-                    properties.getPrincipalClaimName(), properties.getClaimToAuthorityPrefixMap()));
+            .jwtAuthenticationConverter(jwtAuthenticationConverter());
         // @formatter:off
     }
 
+    private Converter<Jwt, AbstractAuthenticationToken> jwtAuthenticationConverter() {
+        JwtAuthenticationConverter converter = new JwtAuthenticationConverter();
+        if (StringUtils.hasText(properties.getPrincipalClaimName())) {
+            converter.setPrincipalClaimName(properties.getPrincipalClaimName());
+        }
+        converter.setJwtGrantedAuthoritiesConverter(
+            new AadJwtGrantedAuthoritiesConverter(properties.getClaimToAuthorityPrefixMap()));
+        return converter;
+    }
```

Comment:
Can the `AadJwtBearerTokenAuthenticationConverter` be replaced by `AadJwtGrantedAuthoritiesConverter`?

Reply:
What's the difference between `BearerTokenAuthentication` and `JwtAuthenticationToken`?

Reply:
Yes, the user can choose to use either. Just if you want to use the `JWT_BEARER` authorization type, you must use `JwtAuthenticationConverter`, because `JwtBearerOAuth2AuthorizedClientProvider` will check that the principal of the obtained token must be Jwt, 

```java
if (!(context.getPrincipal().getPrincipal() instanceof Jwt)) {
	return null;
}
```

There is not a particularly clear statement of their difference, here is a good explanation:
https://stackoverflow.com/a/40375745/6463447
- JWTs are a convenient way to encode and verify claims.
- A Bearer token is just string, potentially arbitrary, that is used for authorization.

Reply:
I moved the `ResourceServerWithOboCondition` to the test source folder to make sure of test coverage.

Reply:
Will remove all the unused conditions.